cmds/identityd/main.go

```diff
@@ -29,63 +38,48 @@ func main() {
 
 	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
 
-	nodeID, err := loadIdentity()
+	server, err := zbus.NewRedisServer(module, msgBrokerCon, workers)
 	if err != nil {
-		os.Exit(1)
+		log.Fatal().Err(err).Msg("failed to create zbus server")
 	}
 
-	farmID, err := identity.GetFarmID()
+	manager, err := identity.NewManager(seedPath)
 	if err != nil {
-		log.Error().Err(err).Msg("fail to read farmer id from kernel parameters")
-		os.Exit(1)
-	}
-
-	store := identity.NewHTTPIDStore(tnodbURL)
-	f := func() error {
-		log.Info().Msg("start registration of the node")
-		if err := store.RegisterNode(nodeID, farmID); err != nil {
-			log.Error().Err(err).Msg("fail to register node identity")
-			return err
-		}
-		return nil
+		log.Fatal().Err(err).Msg("failed to create identity manager")
 	}
 
-	err = backoff.Retry(f, backoff.NewExponentialBackOff())
+	farmID, err := manager.FarmID()
 	if err != nil {
-		return
+		log.Fatal().Err(err).Msg("fail to read farmer id from kernel parameters")
```

Comment:
Question, should we allow a node to be used when it doesn't have a farm ID ?

Reply:
It seems a node can't register if it doesn't have a farmer id set. if the node is not registered, there is no way u can provision anything on it, and u don't have any other means to access it. so it will be useless. 
So I would say no.

Reply:
Fair enough. I created an issue https://github.com/threefoldtech/zosv2/issues/new
So we don't forget to document and enforce this everywhere and be as explicit about in our error message to avoid to much confusion from farmers